archive/tar_writer_factory.go

```diff
@@ -0,0 +1,24 @@
+package archive
+
+import (
+	"archive/tar"
+	"io"
+)
+
+type TarWriter interface {
```

Comment:
We could rename it, but I worry about the implications it would have, though mostly aesthetic. Pack uses this interface in tar-writing code that is not necessarily related to layers. Now maybe Pack could have its own interface that mitigates this, but I'll need to dig more to see if it would compile, given the interface for the factory calls for the use of this interface (however, if that factory interface goes away and is defined by the consumer, we might be able to do this). I'll just try it and see where I land.

Reply:
Resolved. Moving out of this library altogether, along with the factories.